Cython/Compiler/PyrexTypes.py

```diff
@@ -2003,7 +2003,7 @@ def __lt__(self, other):
 
     def py_type_name(self):
         if self.rank <= 4:
-            return "(int, long)"
```

Comment:
This probably breaks the type check in fused types, removing support for `long` in Py2:
https://github.com/cython/cython/blob/236e705d644247415baacf0a6d38b6e85379cce1/Cython/Compiler/FusedNode.py#L296-L304

There are other usages of the method, but this is the only one that I could find that breaks. Thus, special-casing `int` in the fused types type checking code seems a reasonable fix. It would also make the `py_type_name()` method cleaner again.

Reply:
I had the suspicion I was breaking something. I did my best to address this issue. Let's continue the discussion in the other two comments I added after my changes. I'm resolving this one.